### PR TITLE
Add devcontainer configuration

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

Added a simple devcontainer configuration

We already have everything needed to work inside a container, with this configuration file developing in moby is even easier: the IDE will ask you if you want to run everything inside a container and set it up for you. No need to know that you have to run "BIN_DIR=. make shell" any more.

**- How I did it**

**- How to verify it**

Open the repository in Vscode, it will guide you from there

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "moby",
+  "build": {
+    "context": "..",
+    "dockerfile": "../Dockerfile",
+    "target": "dev"
+  },
+  "workspaceFolder": "/go/src/github.com/moby/moby",
+  "workspaceMount": "source=${localWorkspaceFolder},target=/go/src/github.com/moby/moby,type=bind,consistency=cached",
```
Comment (thaJeztah) [inline .devcontainer/devcontainer.json:9]: Might want to use docker/docker for now (until we renamed the code)

Comment (rumpl) [reply .devcontainer/devcontainer.json:9]: Yeah, need to change this, the IDE goes crazy since it doesn't know that there is a redirect

**Comment by rumpl** [comment]
I opened the `cmd/dockerd/daemon.go` file, created a breakpoint and clicked the "Run and Debug" button and:
Screenshot 2023-12-13 at 13 21 12

So it can work out of the box for simple cases, I know that VSCode also has a `launch.json` file you can create to configure how and what you want to debug.

**Review by vvoland — APPROVED** [review]
Nice, LGTM!

**Review by laurazard — APPROVED** [review]
LGTM! :)

**Review by akerouanton — APPROVED** [review]
Dev containers seem broken on GoLand 2023.2.5, so I can't tell whether dlv sessions are working well.

> Failed to deploy 'Dev Container': Cannot invoke "java.io.File.getName()" because "file" is nullCannot invoke "java.io.File.getName()" because "file" is null

But that's a nice addition for newcomers. Thanks!

**Review by thaJeztah — APPROVED** [review]
looks like we have more than enough LGTM's; let's bring this in

**Pull request merged**